lms/djangoapps/completion/api/v1/views.py

```diff
@@ -112,41 +100,27 @@ def post(self, request, *args, **kwargs):
 
         **Returns**
 
-        * If successful.
-         {
-            "message" : _("ok")
-        }
-        * If error the following is returned
-         {
-            "error_code": "error_code" # listed below
-            "error_message" : _("Verbose, plain language description of the problem for
-                                    the app developer with hints about how to fix it.",)
-            "user_message":"Pass this message on to the app user if needed.",
+        A Response object, with an appropriate status code.
+
+        If successful, status code is 200.
+        {
+           "detail" : _("ok")
         }
 
-        * Error Codes
-        * 400 Bad Request
-        *
+        Otherwise, a 400 or 404 may be returned, and the "detail" content will explain the error.
 
         """
         batch_object = request.data or {}
         try:
             user, course_key, blocks = self._validate_and_parse(batch_object)
-        except (ValueError, ValidationError, InvalidKeyError) as e:
+        except ValidationError as exc:
             return Response({
-                "error_message": str(e),
-                "error_code": status.HTTP_400_BAD_REQUEST,
+                "detail": exc.message,
```

Comment:
I used "detail" here because this is what the `IsStaffOrOwner` and `IsAuthenticated` permissions classes use to indicate their errors.

Reply:
Thanks for letting me know. That is definitely a much better solution than what was written in openedx API docs.